Feat: Convert modmail to forum channel, other changes

WhoIsConch opened the pull request:
Converted modmail system to use a forum channel. Cleaned up the code.

## Summary
This pull request changes the ModMail system to use forum channels instead of text channels with threads.

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [x] If code changes were made then they have been tested.
- [x] This PR fixes an issue.

Commit:
Feat: Convert modmail to forum channel, other changes

Converted modmail system to use a forum channel. Cleaned up the code.

WhoIsConch commented:
Oh yeah I also bumped the version of Dpy to latest

iayushanand commented on cogs/modmail.py:
```diff
@@ -1,55 +1,90 @@
-            self.channel = self.bot.get_channel(MODMAIL_CHANNEL_ID)
+        if not self.channel:  
+            self.channel: discord.ForumChannel = self.bot.get_channel(MODMAIL_CHANNEL_ID)
+            print(self.channel.available_tags)
```
seems like useless print message

WhoIsConch replied:
True. Will update

Commit:
Fix: Remove useless print, replace webhook URL

iayushanand reviewed (approved):
lgtm

andrewthederp reviewed (commented):
Why are you fetching the webhook like that instead of `webhook = thread.parent.webhooks[0]`

WhoIsConch commented:
> Why are you fetching the webhook like that instead of `webhook = thread.parent.webhooks[0]`

Good idea. I'll change that up quick.

Commit:
Refactor: Use `thread.parent.webhooks`

This change uses `thread.parent.webhooks` instead of using the webhook URL through aiohttp.

Commit:
Fix: Syntax issue with webhooks method
